packages/autoformat/src/types.ts

```diff
@@ -33,7 +33,8 @@ export interface AutoformatCommonRule {
    */
   query?: (
     editor: TEditor,
-    rule: Omit<AutoformatCommonRule, 'query'>
+    rule: Omit<AutoformatCommonRule, 'query'>,
+    text: string
```

Comment:
Can you merge `text` into the `rule` interface? We generally try to not have more than one parameter object (`editor` being an exception) so it's easier to extend without breaking changes.

Reply:
@zbeyens I've made a change for this, but not 100% sure its what you wanted, can you take a look and let me know. The flexibility of TS interfaces is a bit new to me.

Reply:
Super, I've pushed a new interface `AutoformatQueryOptions` so it's clearer